CharlieBackend.Business/Services/MentorService.cs

```diff
@@ -180,5 +181,41 @@ public async Task<Result<IList<MentorDto>>> GetAllActiveMentorsAsync()
 
             return Result<IList<MentorDto>>.GetSuccess(mentors);
         }
+
+        public async Task<Result<IList<MentorStudyGroupsDto>>> GetMentorStudyGroupsByMentorIdAsync(long id)
+        {
+            if (!await _unitOfWork.MentorRepository.IsEntityExistAsync(id))
+            {
+                return Result<IList<MentorStudyGroupsDto>>.GetError(ErrorCode.NotFound, "Mentor doesn`t exist");
+            }
+
+            var foundGroups = await _unitOfWork.StudentGroupRepository.GetMentorStudyGroups(id);
+
+            if (!foundGroups.Any())
+            {
+                return Result<IList<MentorStudyGroupsDto>>.
+                    GetSuccess(new List<MentorStudyGroupsDto>());
+            }
+
+            return Result<IList<MentorStudyGroupsDto>>.GetSuccess(foundGroups);
+        }
+
+        public async Task<Result<IList<MentorCoursesDto>>> GetMentorCoursesByMentorIdAsync(long id)
+        {
+            if (!await _unitOfWork.MentorRepository.IsEntityExistAsync(id))
+            {
+                return Result<IList<MentorCoursesDto>>.GetError(ErrorCode.NotFound, "Mentor doesn`t exist");
+            }
+
+            var foundCourses = await _unitOfWork.CourseRepository.GetMentorCourses(id);
+
+            if (!foundCourses.Any())
+            {
+                return Result<IList<MentorCoursesDto>>.
+                    GetSuccess(new List<MentorCoursesDto>());
+            }
```

Comment:
You don't need this if at all. repository returns empty list if nothing was found, and you can pass it to line 218